core/src/main/scala/cats/syntax/either.scala

```diff
@@ -276,8 +276,9 @@ final class EitherOps[A, B](private val eab: Either[A, B]) extends AnyVal {
 
   def toEitherNel[AA >: A]: EitherNel[AA, B] = leftMap(NonEmptyList.one)
 
-  def raiseOrPure[F[_]](implicit ev: ApplicativeError[F, A]): F[B] =
-    ev.fromEither(eab)
+  @deprecated("use liftTo instead", "2.0.0")
+  def raiseOrPure[F[_]](implicit F: ApplicativeError[F, _ >: A]): F[B] =
```

Comment:
If we're deprecating this, why change the signature? Should we just leave this as is?

Reply:
That's a good question… as far as I could tell, there was no downside to changing it, so I thought I might as well, in case we end up not deprecating it or it comes in handy for someone. 🤷‍♂

As long as the deprecated version is there too, it might also improve discoverability? I stumbled across `liftTo` for this purpose because I asked about `raiseOrPure` in Gitter and @kubukoz said it sounded like `liftTo` with worse types. Maybe better types for `raiseOrPure` makes it easier to find, and then the deprecation warning points you to the canonically-named method? Idk.

Reply:
The downside is that there is a small chance it breaks source for some users. If we deprecate it , there is really no reason to improve as it only encourage users to use it right?

Reply:
It makes sense to be conservative. I reverted the signature change and left the deprecation in place.

I'm a little curious when this change could break source, though… do you happen to know any examples?

Reply:
@bpholt it's probably going to be some extreme edge cases but imagine where there are two `ApplicativeError` instances for `F` in scope,  one of them is `ApplicativeError[F, A]`

Reply:
Again it's not really expected to break some users, that's why we are totally okay with the other changes.